enterprise/server/remote_execution/container/container.go

```diff
@@ -208,35 +239,39 @@ type CommandContainer interface {
 	//
 	// It is approximately the same as calling PullImageIfNecessary, Create,
 	// Exec, then Remove.
-	Run(ctx context.Context, command *repb.Command, workingDir string, creds oci.Credentials) *interfaces.CommandResult
+	Run(ctx context.Context, task *Task) *interfaces.CommandResult
 
 	// IsImageCached returns whether the configured image is cached locally.
-	IsImageCached(ctx context.Context) (bool, error)
+	IsImageCached(ctx context.Context, task *Task) (bool, error)
 
 	// PullImage pulls the container image from the remote. It always
 	// re-authenticates the request, but may serve the image from a local cache
 	// if needed.
-	PullImage(ctx context.Context, creds oci.Credentials) error
+	PullImage(ctx context.Context, task *Task) error
 
 	// Create creates a new container and starts a top-level process inside it
 	// (`sleep infinity`) so that it stays alive and running until explicitly
 	// removed. Note, this works slightly differently than commands like
 	// `docker create` or `ctr containers create` -- in addition to creating the
 	// container, it also puts it in a "ready to execute" state by starting the
 	// top level process.
-	Create(ctx context.Context, workingDir string) error
+	Create(ctx context.Context, task *Task) error
+
 	// Exec runs a command inside a container, with the same working dir set when
 	// creating the container.
 	//
 	// If stdin is non-nil, the contents of stdin reader will be piped to the
 	// stdin of the executed process. If stdout is non-nil, the stdout of the
 	// executed process will be written to the stdout writer rather than being
 	// written to the command result's stdout field (same for stderr).
-	Exec(ctx context.Context, command *repb.Command, stdio *interfaces.Stdio) *interfaces.CommandResult
+	Exec(ctx context.Context, task *Task, stdio *interfaces.Stdio) *interfaces.CommandResult
```

Comment:
Move stdio to Task?

Reply:
I thought it might be a bit confusing to move it to `Task` because `Run` doesn't respect stdio currently (`stdio` was originally added to support persistent workers).

I do think it'd be a cleaner interface if both `Run` and `Exec` respected stdio though - it would let us simplify our implementation of `--debug_stream_command_outputs` as well as make it simpler to implement live-streaming of stdout/stderr to the UI. But I am leaning toward waiting to move this into `Task` until we actually make that refactor.